LoginServletTest and Login.jsp Updated to Include Password

Linda-Yang opened the pull request:
I updated login.jsp to check for existing usernames and to check if the password is correct. 
In LoginServletTest, I updated the existing user test to include tests for password. I also removed two tests:
1) bad username because register should take care of checking for bad username inputs
2) new user because the new login.jsp redirects back to the login screen if the username does not exist

Commit:
LoginServletTest and Login.jsp update

Commit:
Reverting accidental changes to UserStore.java

saveen-sahni commented on src/main/java/codeu/controller/LoginServlet.java:
```diff
@@ -60,26 +60,29 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
-      User user = new User(UUID.randomUUID(), username, null, Instant.now());
-      userStore.addUser(user);
+    else {
+      request.setAttribute("error", "Invalid password.");
```
Should this be invalid username or password?

Linda-Yang replied:
ah yes! it should be username my bad

saveen-sahni commented on src/main/java/codeu/controller/LoginServlet.java:
```diff
@@ -60,26 +60,29 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
   public void doPost(HttpServletRequest request, HttpServletResponse response)
       throws IOException, ServletException {
     String username = request.getParameter("username");
+    String password = request.getParameter("password");
```
Probably worth checking this edge case - what happens if a user doesn't input anything for one of these fields? Does request.getParameter() give null or an empty string? The first possibility could lead to null pointers.

Linda-Yang replied:
getParameter returns a null string if the parameter does not exist, I will add a null check for both parameters to deal with this case

saveen-sahni commented on src/test/java/codeu/controller/LoginServletTest.java:
```diff
@@ -52,45 +52,16 @@ public void testDoGet() throws IOException, ServletException {
-  }
-
-  @Test
-  public void testDoPost_NewUser() throws IOException, ServletException {
```
I think one way to think about this is - could the LoginServlet still have to deal with a "new user" as an input? If yes, we should cover the case with a unit test. For example, lets say I tried to log in with a made up username (without registering). How should the LoginServlet respond?

Linda-Yang replied:
LoginServlet should respond with error! I will add that test into the tester file

Commit:
Added nullcheck to LoginServlet and additional tests to LoginServletTest

Linda-Yang commented:
I updated LoginServlet and LoginServletTest to reflect the comments made:
1) I added a null check in LoginServlet
2) I added 4 new tests to test for nullPassword, nullUsername, newUser, and wrongPassword

lloza1998 reviewed (approved):
Good job :)
